File: fuel.receipt.manager/src/main/java/com/example/fuel/receipt/manager/entities/Vehicle.java
package com.example.fuel.receipt.manager.entities;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.UuidGenerator;

import java.util.UUID;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Entity
@Table(name = "vehicles")
public class Vehicle {
    @Id
    @UuidGenerator(style = UuidGenerator.Style.AUTO)
    private UUID id;

    @Column(nullable = false)
    private String licensePlate;

    private String brand;
    private String model;
    private Integer year;
    private String vin;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    private User user;
}

File: fuel-receipt-manager/types/vehicle.ts
export interface VehicleRequestDTO {
  licensePlate: string;
  brand: string;
  model: string;
  year: number | null;
  vin: string;
}

export interface VehicleResponseDTO {
  id: string;
  licensePlate: string;
  brand: string | null;
  model: string | null;
  year: number | null;
  vin: string | null;
}
